common/lib/client/realtime.js

```diff
@@ -48,11 +53,18 @@ var Realtime = (function() {
 		channel.onMessage(msg);
 	};
 
-	/* called when a transport becomes connected; reattempt attach()
-	 * for channels that may have been inProgress from a previous transport */
+	/* called when a transport becomes connected; reattempt attach/detach
+	 * for channels that are attaching or detaching.
+	 * Note that this does not use inProgress as inProgress is only channels which have already made
+	* at least one attempt to attach/detach */
 	Channels.prototype.onTransportActive = function() {
-		for(var channelId in this.inProgress)
-			this.inProgress[channelId].checkPendingState();
+		var statechangeOp = RealtimeChannel.progressOps.statechange;
```

Comment:
Oops, thanks. (That line was left from an earlier revision where I was checking `inProgress[channeld][statechangeOp]` in this method, rather than just the channel state)